dataplicity/portforward.py

```diff
@@ -138,9 +144,9 @@ def _connect(self):
         # Set the timeout for initial connect, as default is too high
         _socket.settimeout(5.0)
 
-        log.debug('connecting to %s', self.service.url)
+        log.debug('connecting to %s', ':'.join(map(str, self.host_port)))
```

Comment:
Would this not work?

```
log.debug('connecting to %s:%s', *self.host_port)
```

Or perhaps add a `url` attribute...

Reply:
well, in fact the initial revision *did* have the url attribute but when I have seen that it was merely a string of host, port I thought that you would advise against it, in favor of using host_port pair which is already inside the Connection class. Come to think of it, I could move the url property to the Connection class from Service class :D